panacea/grpc_client.go

```diff
@@ -131,96 +147,106 @@ func (c *GrpcClient) RegisterDataValidator(endpoint string, validatorAcc *Valida
 	txBuilder := txConfig.NewTxBuilder()
 
 	address := validatorAcc.GetAddress()
-	dataValidator := &datapooltypes.DataValidator{
-		Address:  address,
-		Endpoint: endpoint,
-	}
-
-	msgRegisterDataValidator := datapooltypes.NewMsgRegisterDataValidator(dataValidator)
-
-	err := txBuilder.SetMsgs(msgRegisterDataValidator)
-	if err != nil {
-		return err
-	}
-
-	privKey := secp256k1.PrivKey{
-		Key: validatorAcc.secp256k1PrivKey.Bytes(),
-	}
-
-	account, err := c.GetAccount(address)
-	if err != nil {
-		return err
-	}
-
-	sequence := account.GetSequence()
-
-	//TODO: Fee will be set in Config.toml in near future, now just hard-coded.
-	fees := cosmostype.NewCoins(cosmostype.NewInt64Coin("umed", 1000000))
-	txBuilder.SetFeeAmount(fees)
-	txBuilder.SetGasLimit(200000)
-
-	sigV2 := signing.SignatureV2{
-		PubKey: privKey.PubKey(),
-		Data: &signing.SingleSignatureData{
-			SignMode:  signing.SignMode_SIGN_MODE_DIRECT,
-			Signature: nil,
-		},
-		Sequence: sequence,
-	}
 
-	err = txBuilder.SetSignatures(sigV2)
-	if err != nil {
-		return nil
-	}
-
-	//TODO: ChainID will be set in Config.toml in near future, it just hard-coded.
-	chainId, err := c.GetChainId()
-	if err != nil {
-		return err
-	}
-
-	accountNumber := account.GetAccountNumber()
-
-	signerData := xauthsigning.SignerData{
-		ChainID:       chainId,
-		AccountNumber: accountNumber,
-		Sequence:      sequence,
-	}
-
-	sigv2, err := txclient.SignWithPrivKey(signing.SignMode_SIGN_MODE_DIRECT, signerData, txBuilder, &privKey, txConfig, sequence)
-	if err != nil {
-		return nil
-	}
-
-	err = txBuilder.SetSignatures(sigv2)
-	if err != nil {
-		return nil
-	}
-
-	txBytes, err := txConfig.TxEncoder()(txBuilder.GetTx())
-	if err != nil {
-		return err
-	}
-
-	ctx, cancel := context.WithTimeout(context.Background(), 6*time.Second)
-	defer cancel()
-
-	newTxClient := txtypes.NewServiceClient(c.conn)
-	resp, err := newTxClient.BroadcastTx(
-		ctx,
-		&txtypes.BroadcastTxRequest{
-			Mode:    txtypes.BroadcastMode_BROADCAST_MODE_BLOCK,
-			TxBytes: txBytes,
-		},
-	)
-	if err != nil {
-		return nil
-	}
+	getRegisteredDataValidator, _ := c.GetRegisteredDataValidator(address)
+
+	var dataValidator datapooltypes.DataValidator
+
+	if getRegisteredDataValidator == dataValidator {
```

Comment:
When you write `if` statements, it's generally better to handle a shorter case. In this case, you can write:
```go
registeredDataVal, err := c.GetRegisteredDataValidator(address)
if err != nil {
    ...
}

if registeredDataVal != nil {
    // DO NOT return an error if you aren't gonna handle it proper outside of this function.
    // In this case, it's okay to return nil. Then, please put this info to the function comment.
    return nil
}

// Here, you register the dataval to Panacea
```

To do this, you need to change the function signature of `GetRegisteredDataValidator()` to return `*datapooltypes.DataValidator`, not `datapooltypes.DataValidator`.